lib/lightspeed/client.rb

```diff
@@ -26,25 +29,38 @@ def load_json(json)
     end
 
     def get(**args)
-      request(args.merge(method: :get)).perform
+      perform_request(:get, 1, **args)
     end
 
     def post(**args)
-      request(args.merge(method: :post)).perform
+      perform_request(:post, 10, **args)
     end
 
     def put(**args)
-      request(args.merge(method: :put)).perform
+      perform_request(:put, 10, **args)
     end
 
     def delete(**args)
-      request(args.merge(method: :delete)).perform
+      perform_request(:delete, 10, **args)
     end
 
     private
 
+    def perform_request(method, units, **args)
```

Comment:
alright. 

Perhaps instead, have Client.initialise also initialize a RequestQueue, which handles all the buckets and extracts the response etc, and tells the requests when they can go, etc. basically, the methods you have here, but somewhere else. 

I already think the client is doing disparate things by both triggering requests and holding auth stuff, I'd rather not add any more.